File: py3/1024/953.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        ht = {}
        for i, ch in enumerate(order):
            ht[ch] = i

        for i in range(len(words) - 1):
            w1, w2 = words[i], words[i + 1]
            l1, l2 = len(w1), len(w2)
            for j in range(l1):
                if j >= l2 or ht[w1[j]] > ht[w2[j]]:
                    return False
                if ht[w1[j]] < ht[w2[j]]:
                    break

        return True

File: ts/1024/953.ts
function isAlienSorted(words: string[], order: string): boolean {
  const ht = {}, m = order.length, n = words.length
  for (let i = 0; i < m; ++i) {
    ht[order[i]] = i
  }

  for (let i = 1; i < n; ++i) {
    const w1 = words[i - 1],
          w2 = words[i],
          l1 = w1.length,
          l2 = w2.length
    for (let j = 0; j < l1; ++j) {
      if (j >= l2 || ht[w1[j]] > ht[w2[j]]) return false
      if (ht[w1[j]] < ht[w2[j]]) break
    }
  }

  return true
};